ME0 in STA muon reco for 80X

dnash86 opened the pull request:
This PR includes all the code for inclusion of ME0 in STA muon reco.  It has been tested with "runTheMatrix.py --job-reports -j 8 -s -l 1306.0,101.0,1003.0,50202.0,9.0,25202.0", and some output root files were checked for equivalence with an unmodified IB.  Please check for any changes that need to be made and any additional tests that should be performed.  

Thanks a lot.

archiesharma committed:
Porting in 75X for GEMs

dnash86 committed:
Merged Portingin75X from repository archiesharma

dnash86 committed:
First commit of ported modules

dnash86 committed:
DetLayers ported

dnash86 committed:
MeasurmentDet

dnash86 committed:
More packages ported

dnash86 committed:
Final RecoMuon packages

dnash86 committed:
RecoLocalMuon GEMRecHit added

dnash86 committed:
Remaining packages ported

dnash86 committed:
Looking for other hltL2MuonSeeds sources

dnash86 committed:
Commit straggling packages

dnash86 committed:
Fix conflicts

dnash86 committed:
Missing DataFormats, other missing files

dnash86 committed:
Experimenting with me0Customs

dnash86 committed:
Work from pre4

dnash86 committed:
GEMDigitizer customizer

dnash86 committed:
Tracker, RecoVertex changes necessary for customs run

dnash86 committed:
Remaining missing changes

dnash86 committed:
Test files

dnash86 committed:
Experimenting with test config

dnash86 committed:
Other muon elements

dnash86 committed:
Debugging, adding missing functionality

dnash86 committed:
Debugging, adding missing functionality

dnash86 committed:
Experimenting with test config

dnash86 committed:
Experimenting with test config

dnash86 committed:
Continued ME0 investigation

dnash86 committed:
Remove spurious test files

dnash86 committed:
clean root file

dnash86 committed:
Continued ME0 investigation

dnash86 committed:
fix merge

dnash86 committed:
Remove spurious cout statements

dnash86 committed:
Remove accidentally added ouput files

dnash86 committed:
standAloneMuons config with ME0 off

dnash86 committed:
Revent changes for ME0Muons

dnash86 committed:
Revert TrackDetectorAssociator changes as well

dnash86 committed:
Fix failing packages

dnash86 committed:
Delete me0Segments_cfi.py

dnash86 committed:
Revert changes to RecoVertex_cff

dnash86 committed:
Revert changes to aging.py

dnash86 committed:
Finish reverting RecoVertex_cff changes

dnash86 committed:
Remove whitespace from aging.py

dnash86 committed:
Revert changes to combinedCustoms.py

dnash86 committed:
Revert changes to StandAloneMuonAnalyzer.cc

dnash86 committed:
Revert changes to StandAloneMuonAnalyzer.h

dnash86 committed:
Revert changes to StandAloneMuon_cfg.py

dnash86 committed:
Remove whitespace from STAMuonAnalyzer.h

dnash86 committed:
Remove whitespace from STAMuonAnalyzer.h

dnash86 committed:
Remove whitespace from STAMuonAnalyzer.h

dnash86 committed:
Remove whitespace from STAMuonAnalyzer.h

dnash86 committed:
Remove whitespace from STAMuonAnalyzer.h

dnash86 committed:
Remove accidental changes to HLT_FULL

dnash86 committed:
Revert BuildFile.xml

dnash86 committed:
Revert runStandAloneMuon_cfg.py

dnash86 committed:
Remove std::endl

dnash86 committed:
Remove straggling segment code from GEMRecHit

dnash86 committed:
Remove accidental changes to hlt_data

dnash86 committed:
Revert STAMuonAnalyzer

dnash86 committed:
Revert STAMuonAnalyzer

dnash86 committed:
Revert ME0DetId.CC

dnash86 committed:
Revert testME0DetId.cc

dnash86 committed:
Merged ME0DetId.h

dnash86 committed:
Remove repeated line

cmsbuild commented:
A new Pull Request was created by  @dnash86  for CMSSW_8_0_X.

It involves the following packages:

DQM/DTMonitorModule
DQMOffline/JetMET
DataFormats/GEMDigi
DataFormats/MuonDetId
DataFormats/MuonReco
DataFormats/TrackReco
Geometry/GEMGeometry
Geometry/GEMGeometryBuilder
RecoLocalMuon/GEMRecHit
RecoLocalMuon/GEMSegment
RecoMET/METProducers
RecoMuon/CosmicMuonProducer
RecoMuon/DetLayers
RecoMuon/GlobalTrackingTools
RecoMuon/L2MuonProducer
RecoMuon/L3MuonProducer
RecoMuon/MeasurementDet
RecoMuon/MuonIdentification
RecoMuon/MuonSeedGenerator
RecoMuon/Navigation
RecoMuon/StandAloneMuonProducer
RecoMuon/StandAloneTrackFinder
RecoMuon/TrackingTools
RecoMuon/TransientTrackingRecHit
SLHCUpgradeSimulations/Configuration
TrackingTools/TrackFitters


@civanch, @Dr15Jones, @cvuosalo, @ianna, @mdhildreth, @cmsbuild, @deguio, @slava77, @vanbesien, @davidlange6 can you please review it and eventually sign? Thanks.
@ghellwig, @TaiSakuma, @abbiendi, @rappoccio, @Martin-Grunewald, @mmarionncern, @battibass, @ahinzmann, @jhgoh, @dgulhan, @jdolen, @cerati, @trocino, @rociovilar, @GiacomoSguazzoni, @rovere, @VinInn, @bellan, @nhanvtran, @schoef, @amagitte, @makortel, @gpetruc, @istaslis, @mariadalfonso, @bachtis this is something you requested to watch as well.
@slava77, @degano, @smuzaffar you are the release manager for this.

Following commands in first line of a comment are recognized

* **+1|approve[d]|sign[ed]**: L1/L2's to approve it
* **-1|reject[ed]**: L1/L2's to reject it
* **assign &lt;category&gt;[,&lt;category&gt;[,...]]**: L1/L2's to request signatures from other categories
* **unassign &lt;category&gt;[,&lt;category&gt;[,...]]**: L1/L2's to remove signatures from other categories
* **hold**: L1/all L2's/release manager to mark it as on hold
* **unhold**: L1/user who put this PR on hold
* **merge**: L1/release managers to merge this request
* **[@cmsbuild,] please test**: L1/L2 and selected users to start jenkins tests
* **[@cmsbuild,] please test with cms-sw/cmsdist#&lt;PR&gt;**: L1/L2 and selected users to start jenkins tests using externals from cmsdist

slava77 commented on DataFormats/GEMDigi/BuildFile.xml:
```diff
@@ -1,5 +1,8 @@
 <use   name="DataFormats/Common"/>
+<use   name="DataFormats/GeometryVector"/>
 <use   name="DataFormats/MuonDetId"/>
+<use   name="DataFormats/TrackingRecHit"/>
```
why would digi depend on TrackingRecHit?
please explain or remove

slava77 commented on DataFormats/MuonReco/src/classes.h:
```diff
@@ -33,7 +33,7 @@
 #include <map>
 
-namespace DataFormats_MuonReco {
+namespace {
```
what was wrong with this?
all DataFormats/* classes are in their "localized" namespace

slava77 commented on DataFormats/TrackReco/interface/HitPattern.h:
```diff
@@ -34,6 +34,7 @@
 // | mu = 0 |    CSC = 2    | 4*(stat-1)+(ring-1)       |                 | hit type = 0-3 |
 // | mu = 0 |    RPC = 3    | 4*(stat-1)+2*layer+region |                 | hit type = 0-3 |
 // | mu = 0 |    GEM = 4    | 2*(stat-1)+2*(layer-1)    |                 | hit type = 0-3 |
+// |mu = 0  |    ME0 = 5    |                           |                 | hit type = 0-3 |
```
please document the layer derivation logic here for ME0

slava77 commented on DataFormats/TrackReco/interface/HitPattern.h:
```diff
@@ -753,15 +779,33 @@ inline uint16_t HitPattern::getRPCregion(uint16_t pattern)
+  // inline uint32_t HitPattern::getME0Roll(uint32_t pattern) {
+  //   uint32_t sss = getSubSubStructure(pattern), stat = sss >> 1;
+  //   return stat + 1;
+  // }
```
please remove commented out code

slava77 commented on DataFormats/TrackReco/interface/HitPattern.h:
```diff
@@ -753,15 +779,33 @@ inline uint16_t HitPattern::getRPCregion(uint16_t pattern)
+  // }
+  
+  inline uint32_t HitPattern::getME0Layer(uint32_t pattern) {
+    return (getSubSubStructure(pattern) & 1) ;
```
this is not consistent with the format packed in the encodeMuonLayer (returns roll number).
If you don't know what to save, just not save anything (and maybe have no function either)

slava77 commented on DataFormats/TrackReco/src/classes_def.xml:
```diff
@@ -29,7 +29,7 @@
       target="hitPattern"
       version="[-11]"
-      include="utility,DataFormats/SiPixelDetId/interface/PXBDetId.h,DataFormats/SiPixelDetId/interface/PXFDetId.h,DataFormats/SiStripDetId/interface/TIBDetId.h,DataFormats/SiStripDetId/interface/TIDDetId.h,DataFormats/SiStripDetId/interface/TOBDetId.h,DataFormats/SiStripDetId/interface/TECDetId.h,DataFormats/MuonDetId/interface/DTLayerId.h,DataFormats/MuonDetId/interface/CSCDetId.h,DataFormats/M ...
+      include="utility,DataFormats/SiPixelDetId/interface/PXBDetId.h,DataFormats/SiPixelDetId/interface/PXFDetId.h,DataFormats/SiStripDetId/interface/TIBDetId.h,DataFormats/SiStripDetId/interface/TIDDetId.h,DataFormats/SiStripDetId/interface/TOBDetId.h,DataFormats/SiStripDetId/interface/TECDetId.h,DataFormats/MuonDetId/interface/DTLayerId.h,DataFormats/MuonDetId/interface/CSCDetId.h,DataFormats/M ...
```
remove modifications in this ioread block : there was no ME0 in versions up to 11.

slava77 commented on SLHCUpgradeSimulations/Configuration/python/combinedCustoms.py:
```diff
@@ -13,13 +13,13 @@
 
 def cust_2017(process):
-    process=customisePostLS1(process)
+    process=customisePostLS1(process,displayDeprecationWarning=False)
```
just a note: this is not a real change.
github web got confused with the merges

slava77 commented:
@dnash86 
please do "git rebase -i" 
and squash all unnecessary intermediate steps of commits;
it would be best to have the commit where .root  file was added and then removed to end up squashed together so that the .root file doesn't stay in history

dnash86 commented:
@slava77 Thanks for the comments, most of them seem very straightforward.  A question about the .root file removal from history with "git rebase -i": I'm not quite sure how to squash the intermediate commits to remove it.  Do I find the git commit tag from one step before the root file accidental addition, git rebase -i with that tag, then change every operation between addition and removal to "squash"?

slava77 commented:
On 12/15/15 8:45 AM, dnash86 wrote:
> @slava77 <https://github.com/slava77> Thanks for the comments, most of
> them seem very straightforward. A question about the .root file removal
> from history with "git rebase -i": I'm not quite sure how to squash the
> intermediate commits to remove it. Do I find the git commit tag from one
> step before the root file accidental addition, git rebase -i with that
> tag, then change every operation between addition and removal to "squash"?

yes.
Assuming these are close enough and on the same line.
If they are separated by a merge, maybe not really worth it.


> 
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/12796#issuecomment-164822879>.
>

slava77 commented:
BTW, I'm not done with comments.
I'll post a message once I'm done.

dnash86 commented:
Okay, thanks.  In the meantime I'll look in to the rebase.  It may not really be worth it, as you say, I remember having some difficulty removing the root file in the first place.

slava77 commented:
still, please squash the trivial ones (like empty line removals etc)

slava77 commented on Geometry/GEMGeometry/interface/ME0Geometry.h:
```diff
@@ -6,15 +6,18 @@
 
-class GeomDetType;
+//class GeomDetType;
+//class GeomDetUnit;
```
please remove instead of commenting out

slava77 commented on Geometry/GEMGeometryBuilder/src/ME0GeometryBuilderFromCondDB.cc:
```diff
@@ -90,7 +90,48 @@ ME0Geometry* ME0GeometryBuilderFromCondDB::build(const RecoIdealGeometry& rgeo)
+    
+  }
+  /*
+  // Create the ME0Chambers and store them on the Geometry 
```
This looks useful, why can't this be commented out?
Otherwise, remove

slava77 commented on Geometry/GEMGeometryBuilder/src/ME0GeometryBuilderFromDDD.cc:
```diff
@@ -138,5 +155,55 @@ ME0Geometry* ME0GeometryBuilderFromDDD::buildGeometry(DDFilteredView& fview, con
+  auto& partitions(geometry->etaPartitions());
+  // build the chambers and add them to the geometry
+  std::vector<ME0DetId> vDetId;
+  vDetId.clear();
```
looks redundant

slava77 commented on RecoMET/METProducers/interface/CSCHaloDataProducer.h:
```diff
@@ -151,6 +154,9 @@ class CSCHaloDataProducer : public edm::stream::EDProducer<> {
     MuonSegmentMatcher *TheMatcher;
 
+    //Muon Service
+    MuonServiceProxy* TheService;
```
This was removed in #10482

If it's just a forward port artifact, please remove here

slava77 commented on RecoMET/METProducers/src/CSCHaloDataProducer.cc:
```diff
@@ -39,7 +39,7 @@ CSCHaloDataProducer::CSCHaloDataProducer(const edm::ParameterSet& iConfig)
   edm::ParameterSet matchParameters = iConfig.getParameter<edm::ParameterSet>("MatchParameters");
   edm::ConsumesCollector iC = consumesCollector();
-  TheMatcher = new MuonSegmentMatcher(matchParameters, iC);
+  TheMatcher = new MuonSegmentMatcher(matchParameters, TheService, iC);
```
modifications should not be needed

slava77 commented on RecoMuon/CosmicMuonProducer/src/CosmicMuonLinksProducer.cc:
```diff
@@ -80,7 +80,7 @@ CosmicMuonLinksProducer::produce(Event& iEvent, const EventSetup& iSetup)
   for(std::vector<std::pair<edm::EDGetTokenT<reco::TrackCollection>,edm::EDGetTokenT<reco::TrackCollection> > >::const_iterator iLink = theTrackLinks.begin();
      iLink != theTrackLinks.end(); iLink++){
-    LogDebug(category_) << "making map between " << (*iLink).first<<" and "<< (*iLink).second;
+    //LogDebug(category_) << "making map between " << (*iLink).first<<" and "<< (*iLink).second;
```
why comment out a LogDebug?
remove it if it's not needed.

slava77 commented on RecoMuon/CosmicMuonProducer/src/CosmicMuonTrajectoryBuilder.cc:
```diff
@@ -48,6 +48,7 @@ CosmicMuonTrajectoryBuilder::CosmicMuonTrajectoryBuilder(const edm::ParameterSet
   bool enableDTMeasurement = par.getParameter<bool>("EnableDTMeasurement");
   bool enableCSCMeasurement = par.getParameter<bool>("EnableCSCMeasurement");
   bool enableRPCMeasurement = par.getParameter<bool>("EnableRPCMeasurement");
+//bool enableGEMMeasurement = par.getParameter<bool>("EnableGEMMeasurement");
```
remove this
or actually implement

slava77 commented on RecoMuon/DetLayers/interface/MuonDetLayerGeometry.h:
```diff
@@ -83,20 +107,33 @@ class MuonDetLayerGeometry : public DetLayerGeometry{
 
   /// return all backward (-Z) layers (CSC+RPC+GEM), inside-out
   const std::vector<const DetLayer*>& allBackwardLayers() const;
```
why is this block commented out? (it's a duplicate of the implementation above).
please remove or uncomment and rename/implement as described

slava77 commented on RecoMuon/DetLayers/src/MuonDetLayerGeometry.cc:
```diff
@@ -49,18 +52,41 @@ void MuonDetLayerGeometry::addCSCLayers(const pair<vector<DetLayer*>, vector<Det
     gemLayers_fw.push_back(it);
     allForward.push_back(it);
     detLayersMap[ makeDetLayerId(it) ] = it;
+  LogDebug("Muon|RecoMuon|MuonDetLayerGeometry") << "Adding GEMforward "<<std::endl;
```
better preserve the indentation

slava77 commented on RecoMuon/DetLayers/src/MuonDetLayerGeometry.cc:
```diff
@@ -49,18 +52,41 @@ void MuonDetLayerGeometry::addCSCLayers(const pair<vector<DetLayer*>, vector<Det
+}   
+
+
+void MuonDetLayerGeometry::addME0Layers(pair<vector<DetLayer*>, vector<DetLayer*> > me0layers) {
```
pass a const reference, don't copy

slava77 commented on RecoMuon/DetLayers/src/MuonDetLayerGeometry.cc:
```diff
@@ -49,18 +52,41 @@ void MuonDetLayerGeometry::addCSCLayers(const pair<vector<DetLayer*>, vector<Det
+    allForward.push_back(it);
+
+    detLayersMap[ makeDetLayerId(it) ] = it;
+  LogDebug("Muon|RecoMuon|MuonDetLayerGeometry") << "Adding ME0forward "<<std::endl;
```
indent

slava77 commented on RecoMuon/DetLayers/src/MuonDetLayerGeometry.cc:
```diff
@@ -127,7 +153,16 @@ DetId MuonDetLayerGeometry::makeDetLayerId(const DetLayer* detLayer) const{
+      LogDebug("Muon|RecoMuon|MuonDetLayerGeometry") << "Checked "<<std::endl;
+    }
+    return ME0DetId(id.region(),id.layer(),0,0);
+    //return ME0DetId(id.region(),0,0,0);
```
remove?

dnash86 commented:
Hmm, I'm trying the rebase, and got this error:

"error: addinfo_cache failed for path"

for a number of seemingly unrelated files (stuff in IORawData/HcalTBInputService, EventFilter/Utilities, and a few other packages).

Any idea how to fix?  I'm not sure I can just "git rebase --skip" it, since it says it could not apply the commit.

slava77 commented on RecoMuon/DetLayers/interface/MuonDetLayerGeometry.h:
```diff
@@ -150,14 +204,26 @@ class MuonDetLayerGeometry : public DetLayerGeometry{
+//////////////////////////////
+    
+
+  std::map<DetId,DetLayer*> detLayersMap;
```
is non-const DetLayer required here for some reason?

slava77 commented on RecoMuon/DetLayers/src/MuonDetLayerGeometry.cc:
```diff
@@ -284,10 +361,16 @@ const DetLayer* MuonDetLayerGeometry::idToLayer(const DetId &detId) const{
   else throw cms::Exception("InvalidSubdetId")<< detId.subdetId();
 
-  std::map<DetId,const DetLayer*>::const_iterator layer = detLayersMap.find(id);
+  std::map<DetId,DetLayer*>::const_iterator layer = detLayersMap.find(id);
```
why is the "const" removed?

slava77 commented on RecoMuon/DetLayers/src/MuonGEMDetLayerGeometryBuilder.cc:
```diff
@@ -75,6 +75,7 @@ MuonGEMDetLayerGeometryBuilder::buildLayer(int endcap,vector<int>& rings, int st
   vector<const ForwardDetRing*> frontRings, backRings;
 
 
+//  for (std::vector<int>::iterator ring=rings.begin(); ring!=rings.end();ring++){ 
```
remove?

slava77 commented on RecoMuon/DetLayers/src/MuonGEMDetLayerGeometryBuilder.h:
```diff
@@ -10,6 +10,7 @@
 
 class DetLayer;
 class MuRingForwardDoubleLayer;
+//class MuRodBarrelLayer;
```
remove?

slava77 commented on RecoMuon/DetLayers/src/MuonME0DetLayerGeometryBuilder.cc:
```diff
@@ -0,0 +1,225 @@
+#include <RecoMuon/DetLayers/interface/MuDetRod.h>
+
+#include <Utilities/General/interface/precomputed_value_sort.h>
+#include <Geometry/CommonDetUnit/interface/DetSorting.h>
```
please use double-quotes for all CMSSW files

slava77 commented on RecoMuon/DetLayers/src/MuonME0DetLayerGeometryBuilder.cc:
```diff
@@ -0,0 +1,225 @@
+	//No rings defined for ME0
+	// for(int ring = ME0DetId::minRingId; ring <= ME0DetId::maxRingId; ++ring) {
+	//   rings.push_back(ring);
+	// }
```
is any of this commented out code still needed?
please cleanup

slava77 commented on RecoMuon/DetLayers/src/MuonME0DetLayerGeometryBuilder.cc:
```diff
@@ -0,0 +1,225 @@
+// MuonME0DetLayerGeometryBuilder::buildLayer(int endcap, int layer,
+// 					   vector<int>& chambers,
+// 					   vector<int>& rolls,
+// 					   const ME0Geometry& geo) {
```
cleanup commented out code

slava77 commented on RecoMuon/DetLayers/src/MuonME0DetLayerGeometryBuilder.cc:
```diff
@@ -0,0 +1,225 @@
+  //    precomputed_value_sort(muDetRods.begin(), muDetRods.end(), geomsort::ExtractZ<GeometricSearchDet,float>());                                   
+
+
+    //if(backRings.size()!=0 && frontRings.size()!=0) result = new MuRingForwardDoubleLayer(frontRings, backRings);
```
cleanup

slava77 commented:
On 12/15/15 11:04 AM, dnash86 wrote:
> Hmm, I'm trying the rebase, and got this error:
> 
> "error: addinfo_cache failed for path"

addinfo_cache errors can be ignored

> 
> for a number of seemingly unrelated files (stuff in
> IORawData/HcalTBInputService, EventFilter/Utilities, and a few other
> packages).
> 
> Any idea how to fix? I'm not sure I can just "git rebase --skip" it,
> since it says it could not apply the commit.
> 
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/12796#issuecomment-164856804>.
>

dnash86 commented:
@slava77 So do I just "git rebase --skip" through them?  The message seems to imply that the particular commit git is trying to apply during the rebase is not applied, but perhaps "Could not apply cd3924f... Experimenting with test config" does not mean what I interpreted it to mean.

slava77 commented on RecoMuon/MeasurementDet/src/MuonDetLayerMeasurements.cc:
```diff
@@ -76,6 +84,8 @@ MuonRecHitContainer MuonDetLayerMeasurements::recHits(const GeomDet* geomDet,
   theEvent = &iEvent;
   MuonRecHitContainer result;
 
+  //LogDebug("Muon|RecoMuon|MuonDetLayerMeasurements") << "geoId: "<<geoId<<std::endl;
```
please cleanup or uncomment the currently commented out LogDebugs

dnash86 commented:
Also, sorry for messaging mid-review, this conversation can wait till you are done to avoid burying messages in the review if you prefer.

slava77 commented on RecoMuon/MuonIdentification/src/CSCTimingExtractor.cc:
```diff
@@ -82,7 +82,7 @@ CSCTimingExtractor::CSCTimingExtractor(const edm::ParameterSet& iConfig,edm::Con
   edm::ParameterSet matchParameters = iConfig.getParameter<edm::ParameterSet>("MatchParameters");
 
-  theMatcher = new MuonSegmentMatcher(matchParameters, iC);
+  theMatcher = new MuonSegmentMatcher(matchParameters, theService,iC);
```
do not modify

slava77 commented on RecoMuon/MuonIdentification/src/DTTimingExtractor.cc:
```diff
@@ -89,7 +89,7 @@ DTTimingExtractor::DTTimingExtractor(const edm::ParameterSet& iConfig,edm::Consu
   edm::ParameterSet matchParameters = iConfig.getParameter<edm::ParameterSet>("MatchParameters");
 
-  theMatcher = new MuonSegmentMatcher(matchParameters, iC);
+  theMatcher = new MuonSegmentMatcher(matchParameters, theService,iC);
```
do not modify

slava77 commented on RecoMuon/MuonIdentification/test/testME0MuonAnalyzer_fullreco.py:
```diff
@@ -0,0 +1,116 @@
+process.load('Configuration.Geometry.GeometryExtended2023Muon_cff')
+
+
+process.load('Configuration.StandardSequences.MagneticField_38T_PostLS1_cff')
```
MagneticField_38T_PostLS1_cff is deprecated
use MagneticField_cff

slava77 commented on RecoMuon/Navigation/src/MuonNavigationPrinter.cc:
```diff
@@ -55,22 +57,28 @@ MuonNavigationPrinter::MuonNavigationPrinter(const MuonDetLayerGeometry * muonLa
+  else if ( enableCSC & !enableGEM & !enableRPC & !enableME0 ) backward = muonLayout->backwardCSCLayers(); //CSC only
+  else if ( enableCSC & !enableGEM & !enableRPC & enableME0) backward = muonLayout->allCscME0BackwardLayers(); //CSC + ME0
+  else if ( !enableCSC & !enableGEM & !enableRPC & enableME0) backward = muonLayout->backwardME0Layers(); //ME0 only
+  else backward = muonLayout->backwardCSCLayers();
```
this looks wrong: non-upgrade default includes CSC and RPC, which is in this "else" statement if either of the enableGEM or enableME0 are false.

slava77 commented on RecoMuon/Navigation/src/MuonNavigationPrinter.cc:
```diff
@@ -55,22 +57,28 @@ MuonNavigationPrinter::MuonNavigationPrinter(const MuonDetLayerGeometry * muonLa
+  else if ( enableCSC & !enableGEM & !enableRPC & !enableME0 ) forward = muonLayout->forwardCSCLayers(); //CSC only
+  else if ( enableCSC & !enableGEM & !enableRPC & enableME0) forward = muonLayout->allCscME0ForwardLayers(); //CSC + ME0
+  else if ( !enableCSC & !enableGEM & !enableRPC & enableME0) forward = muonLayout->forwardME0Layers(); //ME0 only
+  else forward = muonLayout->forwardCSCLayers();
```
same as above: this looks like a bug

slava77 commented on RecoMuon/StandAloneMuonProducer/python/standAloneMuons_cfi.py:
```diff
@@ -32,10 +32,12 @@
-            GEMRecSegmentLabel = cms.InputTag("gemRecHits"),
             EnableRPCMeasurement = cms.bool(True),
             RPCRecSegmentLabel = cms.InputTag("rpcRecHits"),
+            EnableGEMMeasurement = cms.bool(True),
```
this should be false

slava77 commented on RecoMuon/StandAloneMuonProducer/python/standAloneMuons_cfi.py:
```diff
@@ -56,10 +58,12 @@
-            GEMRecSegmentLabel = cms.InputTag("gemRecHits"),
             EnableRPCMeasurement = cms.bool(True),
             RPCRecSegmentLabel = cms.InputTag("rpcRecHits"),
+            EnableGEMMeasurement = cms.bool(True),
```
false, as before

slava77 commented on RecoMuon/TrackingTools/interface/MuonSegmentMatcher.h:
```diff
@@ -26,7 +26,7 @@ class MuonSegmentMatcher {
 
     /// constructor with Parameter Set and MuonServiceProxy
-  MuonSegmentMatcher(const edm::ParameterSet&, edm::ConsumesCollector& iC);
+  MuonSegmentMatcher(const edm::ParameterSet&, MuonServiceProxy*,edm::ConsumesCollector& iC);
```
do not modify

slava77 commented on RecoMuon/TrackingTools/python/MuonServiceProxy_cff.py:
```diff
@@ -37,8 +37,10 @@
-        GEMLayers = cms.bool(False),
+        CSCLayers = cms.untracked.bool(True),
+        GEMLayers = cms.untracked.bool(True),
+        ME0Layers = cms.bool(True),
```
this should be enabled somewhere else (2019 era or customizations), not in the default location

slava77 commented on RecoMuon/TrackingTools/src/MuonSegmentMatcher.cc:
```diff
@@ -40,7 +41,7 @@ using namespace std;
 // constructors and destructor
 
-MuonSegmentMatcher::MuonSegmentMatcher(const edm::ParameterSet& matchParameters, edm::ConsumesCollector& iC)
+MuonSegmentMatcher::MuonSegmentMatcher(const edm::ParameterSet& matchParameters, MuonServiceProxy* service,edm::ConsumesCollector& iC)
```
do not modify

slava77 commented on RecoMuon/TransientTrackingRecHit/interface/MuonTransientTrackingRecHit.h:
```diff
@@ -8,13 +8,16 @@
 #include "DataFormats/TrackingRecHit/interface/RecSegment.h"
 
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+//#include "FWCore/ServiceRegistry/interface/Service.h"
```
remove?

slava77 commented on SLHCUpgradeSimulations/Configuration/python/aging.py:
```diff
@@ -289,4 +289,3 @@ def ecal_complete_aging_3000(process):
                                     )
 
     return process
-
```
better drop this empty line change so that the file is not on the list of modifications in this PR

slava77 commented on TrackingTools/TrackFitters/src/KFTrajectorySmoother.cc:
```diff
@@ -101,7 +101,7 @@ KFTrajectorySmoother::trajectory(const Trajectory& aTraj) const {
       // std::cout 
                 << "tsos not valid " << currTsos.globalMomentum().perp() << ' ' 
-                << hitSize << ' ' << hitCounter << ' ' << hit->geographicalId() << ' '  
+	//<< hitSize << ' ' << hitCounter << ' ' << hit->geographicalId() << ' '  
```
why is this removed?

slava77 commented on TrackingTools/TrackFitters/src/KFTrajectoryFitter.cc:
```diff
@@ -87,12 +89,12 @@ Trajectory KFTrajectoryFitter::fitOne(const TrajectorySeed& aSeed,
 #ifdef EDM_ML_DEBUG
     if (hit.isValid()) {
-      LogTrace("TrackFitters")<< " ----------------- HIT #" << hitcounter << " (VALID)-----------------------\n"
+      LogDebug("TrackFitters")<< " ----------------- HIT #" << hitcounter << " (VALID)-----------------------\n"
```
@VinInn @rovere 
are you OK with LogDebugs here?

rovere replied:
Not particularly. Why LogTrace was not good enough??

slava77 commented:
On 12/15/15 11:20 AM, dnash86 wrote:
> @slava77 <https://github.com/slava77> So do I just "git rebase --skip"
> through them? The message seems to imply that the particular commit git
> is trying to apply during the rebase is not applied, but perhaps "Could
> not apply cd3924f
> <https://github.com/cms-sw/cmssw/commit/cd3924f2005d3cba0cc90d47c7d345031d97e416>...
> Experimenting with test config" does not mean what I interpreted it to
> mean.

It likely means something else.
What was  the full message?

> 
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/12796#issuecomment-164862664>.
>

slava77 commented:
@cmsbuild please test

David,
I'm done looking at the code.

cmsbuild commented:
The tests are being triggered in jenkins.

dnash86 commented:
The full message is this:



error: addinfo_cache failed for path 'L1Trigger/L1TCalorimeter/python/caloStage1Params_Deprecated_cfi.py'
error: addinfo_cache failed for path 'IORawData/HcalTBInputService/plugins/HcalTBWriter.h'
error: addinfo_cache failed for path 'IORawData/HcalTBInputService/plugins/HcalTBWriter.cc'
error: addinfo_cache failed for path 'IORawData/HcalTBInputService/plugins/HcalTBSource.h'
error: addinfo_cache failed for path 'IORawData/HcalTBInputService/plugins/HcalTBSource.cc'
error: addinfo_cache failed for path 'Geometry/CaloTopology/test/HcalTopologyTester.cc'
error: addinfo_cache failed for path 'FWCore/ServiceRegistry/src/CurrentModuleOnThread.cc'
error: addinfo_cache failed for path 'FWCore/ServiceRegistry/interface/CurrentModuleOnThread.h'
error: addinfo_cache failed for path 'EventFilter/Utilities/src/FedRawDataInputSource.cc'
error: addinfo_cache failed for path 'EventFilter/Utilities/interface/FedRawDataInputSource.h'
error: addinfo_cache failed for path 'DQM/HLTEvF/plugins/FourVectorHLT.h'
error: addinfo_cache failed for path 'Configuration/Geometry/python/GeometryExtended2023DevReco_cff.py'
error: addinfo_cache failed for path 'Configuration/Generator/python/H125GGgluonfusion_13TeV_TuneCUETP8M1_cfi.py'
error: addinfo_cache failed for path 'CalibCalorimetry/EcalLaserAnalyzer/src/classes.h'
error: addinfo_cache failed for path 'CalibCalorimetry/EcalCorrelatedNoiseAnalysisAlgos/src/classes.h'
error: could not apply cd3924f... Experimenting with test config

When you have resolved this problem, run "git rebase --continue".
If you prefer to skip this patch, run "git rebase --skip" instead.
To check out the original branch and stop rebasing, run "git rebase --abort".
Could not apply cd3924f... Experimenting with test config

dnash86 commented:
And thanks for the message saying you're done, I'll start going through your comments soon.

slava77 commented:
what do you get with ``` git status ``` ?
also, are you in the original development branch or someplace else (as after ``` git cms-merge-topic 12796 ```)

dnash86 commented:
Full message from "git status":

# HEAD detached at cc97cbb
# You are currently rebasing branch 'ME0-InFullMuonReco-For80X-Portv2' on 'cc97cbb'.
#   (fix conflicts and then run "git rebase --continue")
#   (use "git rebase --skip" to skip this patch)
#   (use "git rebase --abort" to check out the original branch)
#
# Unmerged paths:
#   (use "git reset HEAD <file>..." to unstage)
#   (use "git add <file>..." to mark resolution)
#
#	both modified:      RecoLocalMuon/GEMRecHit/test/runTestUpgradeGEMME0Reco_cfg.py
#
# Untracked files:
#   (use "git add <file>..." to include in what will be committed)
#
#	RecoLocalMuon/GEMRecHit/test/TestME0SegmentAnalyzer.cc
#	RecoLocalMuon/GEMRecHit/test/testme0segementanalyzer_cfg.py
no changes added to commit (use "git add" and/or "git commit -a")


I'm in a new area, a recent IB:  CMSSW_8_0_X_2015-12-15-1100, after doing git-cms-merge-topic dnash86:ME0-InFullMuonReco-For80X-Portv2.  Should I instead have used the PR number?

slava77 commented:
git merge-topic adds a merge commit and you are no longer in your original ME0-InFullMuonReco-For80X-Portv2. Rebasing in this case will be complicated and you will have problems to update this PR itself.

dnash86 commented:
Ah, okay - should I abort the rebase and just work on the comments, in that case?

slava77 commented:
you may be better off with the following
```
git reset --hard CMSSW_8_0_X_2015-12-15-1100
git fetch my-cmssw ME0-InFullMuonReco-For80X-Portv2:ME0-InFullMuonReco-For80X-Portv2
git checkout ME0-InFullMuonReco-For80X-Portv2
git rebase -i
git rebase CMSSW_8_0_X_2015-12-15-1100
# and then the usual
git cms-checkdeps
scram b -r vclean
scram b
```

dnash86 commented:
Hmm, did you mean for me to do that after the merge-topic?  If so, I actually still get the same error.  Well, I also do "git rebase -i HEAD~X" in place of "git rebase -i" where X is some number before the commits I am trying to squash.

slava77 commented:
it looks like it stumbles over the bunch of merges present in the history already

dnash86 commented:
It seems that way to me, yes.

slava77 commented:
I guess the idea to get rid of root files by squashing can be given up.
The git pack cache will eventually be cleaned up, so the files will not pollute the history for too long.

dnash86 commented:
Okay.  I will be in touch with the edits/responses to your review comments.  Thanks.

cmsbuild commented:
+1
Tested at: 865e21db6182e62e8c43902490e139dce763a1fd
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-12796/10317/summary.html

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-12796/10317/summary.html

dnash86 committed:
Fixes from comments on PR

dnash86 committed:
Fixes from comments on PR

dnash86 commented:
I've addressed the comments now.  A few short responses:

As suggested, I removed a few functions from HitPattern that I wasn't quite sure how to implement.  Although now that means that HitPattern.cc won't print something out in the ME0 case.  

For a few files, I had forgotten to revert changes I had put in during testing, sorry about that.  (MuonServiceProxy,KFTrjacectoryFitter)

Regarding MuonNavigationPrinter:  I've reverted the "else" to work as it did before (going to allForward/allBackward in the "else").

cmsbuild commented:
Pull request #12796 was updated. @civanch, @Dr15Jones, @cvuosalo, @ianna, @mdhildreth, @cmsbuild, @deguio, @slava77, @vanbesien, @davidlange6 can you please check and sign again.

dnash86 commented:
Please let me know if there are any further questions or changes I should implement.

slava77 commented:
https://github.com/dnash86/cmssw/compare/865e21d...14d417d#diff-146dddc7eb1d8ffe88c61e44d53b1187L864
I think just saying "me0" is good without further details.
If the hit is there, I think that it should be printed (better than saying "UNKNOWN Muon SubStructure"),
considering that muonME0HitFilter exists and looks ok.

deguio commented on DQM/DTMonitorModule/python/dtChamberEfficiency_Cosmics_cfi.py:
```diff
@@ -12,6 +12,7 @@
     dt4DSegments = cms.InputTag("dt4DSegments"),
     theRPCRecHits = cms.InputTag("dummy"),
     thegemRecHits = cms.InputTag("dummy"),
+    theME0RecHits = cms.InputTag("dummy"),
```
looks like these parameters are not read from the c++
are they needed?

dnash86 committed:
Second round comments

cmsbuild commented:
Pull request #12796 was updated. @civanch, @Dr15Jones, @cvuosalo, @ianna, @mdhildreth, @cmsbuild, @deguio, @slava77, @vanbesien, @davidlange6 can you please check and sign again.

dnash86 commented:
@slava77 I put the case for me0 back in, just printing out "me0" as you suggested.

@deguio I removed those parameters that are not read.

slava77 commented:
@cmsbuild please test

cmsbuild commented:
The tests are being triggered in jenkins.

deguio commented:
thanks @dnash86 
the same parameters can be removed from the cosmics cfi as well.
for the rest the code can go in (referring to DQM part)
thanks,
F.

dnash86 committed:
Second round comments

cmsbuild commented:
Pull request #12796 was updated. @civanch, @Dr15Jones, @cvuosalo, @ianna, @mdhildreth, @cmsbuild, @deguio, @slava77, @vanbesien, @davidlange6 can you please check and sign again.

dnash86 commented:
@deguio Ah, sorry, missed that one.  Just removed them from the cosmics cfi as well.

deguio commented:
please test

deguio commented:
+1

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
-1 
Tested at: 0832ae6d374d1be7d56b50bd621f8e284c6d1b77
 I found an error when building: 
 
 <pre>>> Compiling edm plugin /tmp/cmsbuild/workspace/ib-any-integration/CMSSW_8_0_X_2015-12-16-2300/src/DQMOffline/JetMET/src/HTMHTAnalyzer.cc 
>> Compiling edm plugin /tmp/cmsbuild/workspace/ib-any-integration/CMSSW_8_0_X_2015-12-16-2300/src/DQMOffline/JetMET/src/JetAnalyzer.cc 
>> Compiling edm plugin /tmp/cmsbuild/workspace/ib-any-integration/CMSSW_8_0_X_2015-12-16-2300/src/DQMOffline/JetMET/src/JetAnalyzer_HeavyIons.cc 
>> Compiling edm plugin /tmp/cmsbuild/workspace/ib-any-integration/CMSSW_8_0_X_2015-12-16-2300/src/DQMOffline/JetMET/src/JetAnalyzer_HeavyIons_matching.cc 
/tmp/cmsbuild/workspace/ib-any-integration/CMSSW_8_0_X_2015-12-16-2300/src/DQMOffline/JetMET/src/BeamHaloAnalyzer.cc: In constructor 'BeamHaloAnalyzer::BeamHaloAnalyzer(const edm::ParameterSet&)':
/tmp/cmsbuild/workspace/ib-any-integration/CMSSW_8_0_X_2015-12-16-2300/src/DQMOffline/JetMET/src/BeamHaloAnalyzer.cc:73:70: error: no matching function for call to 'MuonSegmentMatcher::MuonSegmentMatcher(edm::ParameterSet&, MuonServiceProxy*&, edm::ConsumesCollector&)'
   TheMatcher = new MuonSegmentMatcher(matchParameters, TheService, iC);
                                                                      ^
/tmp/cmsbuild/workspace/ib-any-integration/CMSSW_8_0_X_2015-12-16-2300/src/DQMOffline/JetMET/src/BeamHaloAnalyzer.cc:73:70: note: candidates are:
In file included from /tmp/cmsbuild/workspace/ib-any-integration/CMSSW_8_0_X_2015-12-16-2300/src/DQMOffline/JetMET/interface/BeamHaloAnalyzer.h:118:0,
                 from /tmp/cmsbuild/workspace/ib-any-integration/CMSSW_8_0_X_2015-12-16-2300/src/DQMOffline/JetMET/src/BeamHaloAnalyzer.cc:1:
</pre> 
 you can see the results of the tests here: 
 https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-12796/10346/summary.html

dnash86 committed:
Remove TheService

dnash86 commented:
Looks like we missed one instance where "TheService, " should be removed.  I'll push the removal now.

cmsbuild commented:
Pull request #12796 was updated. @civanch, @Dr15Jones, @cvuosalo, @ianna, @mdhildreth, @cmsbuild, @deguio, @slava77, @vanbesien, @davidlange6 can you please check and sign again.

cmsbuild commented:
-1 
Tested at: 193c2a9da0e00eeedf44c5acc880d260381c3b84
 I found an error when building: 
 
 <pre>>> Compiling edm plugin /tmp/cmsbuild/workspace/ib-any-integration/CMSSW_8_0_X_2015-12-16-2300/src/DQMOffline/JetMET/src/HCALRecHitAnalyzer.cc 
>> Compiling edm plugin /tmp/cmsbuild/workspace/ib-any-integration/CMSSW_8_0_X_2015-12-16-2300/src/DQMOffline/JetMET/src/HTMHTAnalyzer.cc 
>> Compiling edm plugin /tmp/cmsbuild/workspace/ib-any-integration/CMSSW_8_0_X_2015-12-16-2300/src/DQMOffline/JetMET/src/JetAnalyzer.cc 
>> Compiling edm plugin /tmp/cmsbuild/workspace/ib-any-integration/CMSSW_8_0_X_2015-12-16-2300/src/DQMOffline/JetMET/src/JetAnalyzer_HeavyIons.cc 
/tmp/cmsbuild/workspace/ib-any-integration/CMSSW_8_0_X_2015-12-16-2300/src/DQMOffline/JetMET/src/BeamHaloAnalyzer.cc: In constructor 'BeamHaloAnalyzer::BeamHaloAnalyzer(const edm::ParameterSet&)':
/tmp/cmsbuild/workspace/ib-any-integration/CMSSW_8_0_X_2015-12-16-2300/src/DQMOffline/JetMET/src/BeamHaloAnalyzer.cc:73:70: error: no matching function for call to 'MuonSegmentMatcher::MuonSegmentMatcher(edm::ParameterSet&, MuonServiceProxy*&, edm::ConsumesCollector&)'
   TheMatcher = new MuonSegmentMatcher(matchParameters, TheService, iC);
                                                                      ^
/tmp/cmsbuild/workspace/ib-any-integration/CMSSW_8_0_X_2015-12-16-2300/src/DQMOffline/JetMET/src/BeamHaloAnalyzer.cc:73:70: note: candidates are:
In file included from /tmp/cmsbuild/workspace/ib-any-integration/CMSSW_8_0_X_2015-12-16-2300/src/DQMOffline/JetMET/interface/BeamHaloAnalyzer.h:118:0,
                 from /tmp/cmsbuild/workspace/ib-any-integration/CMSSW_8_0_X_2015-12-16-2300/src/DQMOffline/JetMET/src/BeamHaloAnalyzer.cc:1:
</pre> 
 you can see the results of the tests here: 
 https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-12796/10351/summary.html

slava77 commented on RecoMET/METProducers/interface/CSCHaloDataProducer.h:
```diff
@@ -30,6 +30,7 @@
 #include "FWCore/Framework/interface/Event.h"
 #include "FWCore/Framework/interface/MakerMacros.h"
 #include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "RecoMuon/TrackingTools/interface/MuonServiceProxy.h"
```
includes should be dropped as well.
There should be no changes in this file

slava77 commented on RecoMuon/TrackingTools/src/MuonSegmentMatcher.cc:
```diff
@@ -17,6 +17,7 @@
 #include "FWCore/Framework/interface/Event.h"
 
 #include "DataFormats/MuonReco/interface/Muon.h"
+#include "RecoMuon/TrackingTools/interface/MuonServiceProxy.h"
```
there should be no changes in this file

slava77 commented on RecoLocalMuon/GEMRecHit/test/BuildFile.xml:
```diff
@@ -27,7 +27,7 @@
 <use   name="MagneticField/Engine"/>
 <use   name="MagneticField/Records"/>
-
+<use   name="DQMServices/Core"/>
```
this was previously removed by #11327
I think there should be no changes in this file in this PR

slava77 commented on RecoLocalMuon/GEMRecHit/src/SealModule.cc:
```diff
@@ -16,4 +16,3 @@ DEFINE_FWK_MODULE(ME0RecHitProducer);
 
 #include "RecoLocalMuon/GEMRecHit/src/ME0RecHitStandardAlgo.h"
 DEFINE_EDM_PLUGIN (ME0RecHitAlgoFactory, ME0RecHitStandardAlgo, "ME0RecHitStandardAlgo");
-
```
here and in other places it would be better to cleanup spurious empty line changes

slava77 commented on DataFormats/GEMDigi/BuildFile.xml:
```diff
@@ -1,5 +1,7 @@
 <use   name="DataFormats/Common"/>
+<use   name="DataFormats/GeometryVector"/>
 <use   name="DataFormats/MuonDetId"/>
+<use   name="rootrflx"/>
```
this was previously cleaned up in #8090 
https://github.com/cms-sw/cmssw/pull/8090/files#diff-f926bac44e8a8673b0eed81dd1550b8fL4
I think there should be no changes in this file (this is the only change in DataFormats/GEMDigi and we didn't need it to compile so far)

slava77 commented on RecoMuon/DetLayers/plugins/MuonDetLayerGeometryESProducer.cc:
```diff
@@ -12,11 +13,13 @@
 #include <Geometry/CSCGeometry/interface/CSCGeometry.h>
 #include <Geometry/RPCGeometry/interface/RPCGeometry.h>
 #include <Geometry/GEMGeometry/interface/GEMGeometry.h>
+#include <Geometry/GEMGeometry/interface/ME0Geometry.h>
```
still in <>, better use ""

dnash86 replied:
@slava77 Sorry, which lines do you mean?  13-16?

slava77 replied:
I mean all of the includes of CMSSW package files
It's mostly style, not required

dnash86 commented:
I am confused, this error seems to point at the line I just changed, which does not have the "TheService," in it any more?  Am I reading a filename wrong?  It looks like the fix for this has been done, here: https://github.com/dnash86/cmssw/commit/8944457e42eec33fe7f1876a7d3e3784a3516172

dnash86 commented:
Also, will implement those changes in a moment

slava77 commented:
regarding https://github.com/cms-sw/cmssw/pull/12796#issuecomment-165575950
check the commit ID for which the tests were done, they are included with the compilation error message sent by cmsbuild/bot

dnash86 committed:
third round comments

cmsbuild commented:
Pull request #12796 was updated. @civanch, @Dr15Jones, @cvuosalo, @ianna, @mdhildreth, @cmsbuild, @deguio, @slava77, @vanbesien, @davidlange6 can you please check and sign again.

dnash86 commented:
That commit includes the recent comments, excluding the style for include files, but including some removals of other spurious newline modifications that I found.

slava77 commented on DataFormats/GEMDigi/BuildFile.xml:
```diff
@@ -1,4 +1,5 @@
 <use   name="DataFormats/Common"/>
+<use   name="DataFormats/GeometryVector"/>
```
my previous comment to this file related to all changes.
There are no other changes to this package ( DataFormats/GEMDigi). Why does the build file need a change?
If not needed, please remove.

dnash86 committed:
Remove change to BuildFile

cmsbuild commented:
Pull request #12796 was updated. @civanch, @Dr15Jones, @cvuosalo, @ianna, @mdhildreth, @cmsbuild, @deguio, @slava77, @vanbesien, @davidlange6 can you please check and sign again.

dnash86 commented:
Removed that line.

slava77 commented:
@cmsbuild please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
-1 
 
Tested at: 6e5566c584ef0ee2db4792d728c55b8bf92bb4f4
I found errors in the following unit tests: 
 
 ---> test testRecoMETMETProducers had ERRORS
 
 
you can see the results of the tests here: 
 https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-12796/10353/summary.html

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-12796/10353/summary.html

ianna commented:
+1

The error seems unrelated to this PR:
```
18-Dec-2015 01:30:01 CET  Initiating request to open file file:ttbarForMetTests.root
----- Begin Fatal Exception 18-Dec-2015 01:30:01 CET-----------------------
An exception of category 'FileOpenError' occurred while
   [0] Constructing the EventProcessor
   [1] Constructing input source of type PoolSource
   [2] Calling RootFileSequenceBase::initTheFile()
   [3] Calling StorageFactory::open()
   [4] Calling File::sysopen()
Exception Message:
Failed to open the file 'ttbarForMetTests.root'
   Additional Info:
      [a] Input file file:ttbarForMetTests.root could not be opened.
      [b] open() failed with system error 'No such file or directory' (error code 2)
----- End Fatal Exception -------------------------------------------------
Failure using recoMET_caloMet_cfg.py: status 84
status = 21504

---> test testRecoMETMETProducers had ERRORS
 
^^^^ End Test testRecoMETMETProducers ^^^^
```

slava77 commented:
+1

for https://github.com/cms-sw/cmssw/pull/12796	6e5566c
- code changes are OK, after several iterations with comments addressed
- jenkins tests pass and comparisons with the baseline show no differences as expected

deguio commented:
+1

ianna commented:
@cmsbuild please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: 6e5566c584ef0ee2db4792d728c55b8bf92bb4f4
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-12796/10417/summary.html

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-12796/10417/summary.html
